libmamba/include/mamba/specs/package_info.hpp

```diff
@@ -81,4 +82,47 @@ namespace mamba::specs
 
     void from_json(const nlohmann::json& j, PackageInfo& pkg);
 }
+
+template <>
+struct std::hash<mamba::specs::PackageInfo>
+{
+    auto operator()(const mamba::specs::PackageInfo& pkg) const -> std::size_t
+    {
+        auto seed = std::size_t(0);
+        seed = mamba::util::hash_combine_val(seed, pkg.name);
```

Comment:
All this code would be simplified by using `hash_vals` instead (it's the generalisation of the code pattern here, if my understanding is correct)
Same remark for other hash impls.